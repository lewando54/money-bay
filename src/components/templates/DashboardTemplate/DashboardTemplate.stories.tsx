/* eslint-disable @typescript-eslint/consistent-type-assertions */
import { StoryObj, Meta } from '@storybook/react-native'

import DashboardTemplate from './DashboardTemplate'

const meta = {
	component: DashboardTemplate,
	argTypes: {
        // To write
	}
} satisfies Meta<typeof DashboardTemplate>

export default meta
type Story = StoryObj<typeof meta>

export const Default: Story = {
	args: {
        // To write
	}
}
