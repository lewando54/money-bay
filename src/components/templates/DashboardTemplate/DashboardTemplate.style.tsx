import { Dimensions, StyleSheet } from 'react-native'

import globalThemeStyle, { MAIN_BG, MAIN_DARK } from '../../../styling/GlobalTheme.style'

const {width, height} = Dimensions.get('screen')

export default StyleSheet.create({
    mainContainer: {
        flex: 1,
        backgroundColor: '#FAFCFF'
    },
    imageBackgroundContainer: {
        height: height * 0.442118,
        backgroundColor: MAIN_BG,
        borderBottomRightRadius: 20,
        borderBottomLeftRadius: 20,
        marginBottom: height * -0.0307
    },
    headerContainer: {
        flexDirection: 'row',
        justifyContent: 'space-between',
        marginTop: height * 0.0541,
        paddingHorizontal: width * 0.0426,
        borderBottomColor: '#CED6E14E',
        borderBottomWidth: 1,
        paddingBottom: height * 0.0221
    },
    miniProfileIcon: {
        width: 22,
        height: 22
    },
    currencyValue: {
        ...globalThemeStyle.text_Regular,
        fontSize: 16
    },
    cardsList: {
        padding: 20,
        height: height * 0.2660,
        flexGrow: 0
    },
    buttonsAndPaymentsContainer: {
        paddingHorizontal: width * 0.0533,
        gap: height * 0.0309
    },
    buttonsContainer: {
        flexDirection: 'row',
        justifyContent: 'center',
        gap: 23
    },
    latestTransactionContainer: {
        gap: 6
    },
    latestTransactionHeadingContainer: {
        flexDirection: 'row',
        alignItems: 'flex-end',
        justifyContent: 'space-between',
        marginBottom: 6
    },
    latestTransactionHeading: {
        ...globalThemeStyle.text_Medium,
        color: MAIN_DARK,
        fontSize: 20
    },
    latestTransactionViewAllLink: {
        fontSize: 16
    }
})
