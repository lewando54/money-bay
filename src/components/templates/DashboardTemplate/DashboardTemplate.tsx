import React from 'react'
import dashboardTemplateStyle from './DashboardTemplate.style'

interface IDashboardTemplateProps {
    // To write
    testID?: string
}

export default function DashboardTemplate({
    testID
}: IDashboardTemplateProps){
    return (
        <></>
    )
}
