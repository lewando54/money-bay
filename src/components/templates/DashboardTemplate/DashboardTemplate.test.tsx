import React from 'react'
import renderer from 'react-test-renderer'
import { render, screen } from '@testing-library/react-native'
import DashboardTemplate from './DashboardTemplate'
describe('DashboardTemplate', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<DashboardTemplate />)
    expect(tree).toMatchSnapshot()
  })
})
