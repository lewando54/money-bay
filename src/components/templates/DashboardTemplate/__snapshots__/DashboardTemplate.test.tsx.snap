// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardTemplate should render correctly 1`] = `null`;
